src/jit/compiler.cpp

```diff
@@ -383,6 +383,101 @@ histo       loopExitCountTable(DefaultAllocator::Singleton(), loopExitCountBucke
 #endif // COUNT_LOOPS
 
 //------------------------------------------------------------------------
+// getJitGCType: Given the VM's CorInfoGCType convert it to the JIT's var_types
+//
+// Arguments:
+//    gcType    - an enum value that originally came from an element 
+//                of the BYTE[] returned from getClassGClayout()
+//
+// Return Value:
+//    The corresponsing enum value from the JIT's var_types
+//
+// Notes:
+//   The gcLayout of each field of a struct is returned from getClassGClayout()
+//   as a BYTE[] but each BYTE element is actually a CorInfoGCType value
+//   Note when we 'know' that there is only one element in theis array
+//   the JIT will often pass the address of a single BYTE, instead of a BYTE[]
+//
+
+var_types    Compiler::getJitGCType(BYTE gcType)
+{
+    var_types result = TYP_UNKNOWN;
+    CorInfoGCType corInfoType = (CorInfoGCType)gcType;
+
+    if (corInfoType == TYPE_GC_NONE)
+    {
+        result = TYP_I_IMPL;
+    }
+    else if (corInfoType == TYPE_GC_REF)
+    {
+        result = TYP_REF;
+    }
+    else if (corInfoType == TYPE_GC_BYREF)
+    {
+        result = TYP_BYREF;
+    }
+    else
+    {
+        noway_assert(!"Bad value of 'gcType'");
+    }
+    return result;
+}
+
+#if FEATURE_MULTIREG_ARGS
+//------------------------------------------------------------------------
+// getStructGcPtrsFromOp: Given a GenTree node of TYP_STRUCT that represents a pass by value argument
+//                        return the gcPtr layout for the pointers sized fields //
+// Arguments:
+//    op         - the operand of TYP_STRUCT that is passed by value
+//    gcPtrsOut  - an array of BYTES that are written by this method
+//                 they will contain the VM's CorInfoGCType values 
+//                 for each pointer sized field//
+// Return Value:
+//     Two [or more] values are written into the gcPtrs array
+//
+// Note that for ARM64 there will alwys be exactly two pointer sized fields
+
+void Compiler::getStructGcPtrsFromOp(GenTreePtr op, BYTE *gcPtrsOut)
+{
+    assert(op->TypeGet() == TYP_STRUCT);
```

Comment:
Probably, though I think we could also add that it is a precondition that the type must be a non-special SIMD type. At least for now (and probably longer term) any non-TYP_STRUCT struct will have no GC pointers.